public/app/features/alerting/unified/components/rule-editor/labels/LabelsField.tsx

```diff
@@ -67,41 +73,21 @@ function mapLabelsToOptions(items: Iterable<string> = []): Array<SelectableValue
   return Array.from(items, (item) => ({ label: item, value: item }));
 }
 
-const RemoveButton: FC<{
-  remove: (index?: number | number[] | undefined) => void;
-  className: string;
-  index: number;
-}> = ({ remove, className, index }) => (
-  <Button
-    className={className}
-    aria-label="delete label"
-    icon="trash-alt"
-    data-testid={`delete-label-${index}`}
-    variant="secondary"
-    onClick={() => {
-      remove(index);
-    }}
-  />
-);
-
-const AddButton: FC<{
-  append: UseFieldArrayAppend<RuleFormValues, 'labels'>;
-  className: string;
-}> = ({ append, className }) => (
-  <Button
-    className={className}
-    icon="plus-circle"
-    type="button"
-    variant="secondary"
-    onClick={() => {
-      append({ key: '', value: '' });
-    }}
-  >
-    Add label
-  </Button>
-);
-
-const LabelsWithSuggestions: FC<{ dataSourceName: string }> = ({ dataSourceName }) => {
+export const LabelsInRule = () => {
+  const { watch } = useFormContext<RuleFormValues>();
+  const labels = watch('labels');
+
+  const labelsObj: Record<string, string> = labels.reduce((acc: Record<string, string>, label) => {
+    if (label.key) {
```

Comment:
I noticed that it's possible to have keys without value , so we only check the key